lib/web_ui/lib/src/engine/host_node.dart

```diff
@@ -13,7 +14,25 @@ import 'text_editing/text_editing.dart';
 /// (preferred Flutter rendering method) and [DomDocument] (fallback).
 ///
 /// Not to be confused with [DomDocumentOrShadowRoot].
+///
+/// This also handles the stylesheet that is applied to the different types of
+/// HostNodes; for ShadowDOM there's not much to do, but for ElementNodes, the
+/// stylesheet is "namespaced" by the `flt-glass-pane` prefix, so it "only"
+/// affects things that Flutter web owns.
 abstract class HostNode {
+  /// Returns an appropriate HostNode for the given [root].
+  ///
+  /// If `attachShadow` is supported, this returns a [ShadowDomHostNode], else
+  /// this will fall-back to an [ElementHostNode].
+  factory HostNode.create(DomElement root, String defaultFont) {
+    if (getJsProperty<Object?>(root, 'attachShadow') != null) {
```

Comment:
I _think_ that syntax is Bad For Dart2Wasm, [look at this change](https://github.com/flutter/engine/commit/3aa3d2a8f23da96274939b7e494d29be4776d2df#diff-85b3bd302d38fbd6b511f9d2636fee56d38b4969e9b8ced1f144db161e720112L26-R34)

Not exactly sure if it's because we're calling an external function something that we "get" (just to check if it exists) or what. This code already exists in the engine as is, I'm just moving it from the embedder to the host_node file :/

Reply:
Confirmed with the dart2wasm folk that this might be a problematic change, so skipping this one.

Sorry Mouad, but thanks for taking the time to point this improvement out!